File: Arithmetic_HCFInput.java
package arithmetic;

import java.awt.HeadlessException;
import java.util.Scanner;
import java.util.ArrayList;
import java.util.List;
import javax.swing.JOptionPane;

public class Arithmetic {

    public static void main(String[] args) {
        System.out.println("Welcome to our demonstration sequels");
        System.out.println("Hope you enjoy (and follow) the lessons.");
        System.out.println("");

        // Collect input
        String collect_input; // For collecting user input

        /* Scanner takes care of collecting input using the System.in object. */
        Scanner user_choice = new Scanner(System.in);

        FastHCF h_c_f;
        List<Integer> set;
        set = new ArrayList<>();

        System.out.println("Welcome to our Find HCF program.");
        System.out.println("Enter your series of number for HCF.");
        System.out.println("Type 'done' when you are through with entering your numbers.\n");
        System.out.print("Enter First Number:  ");
        try {
            do {
                collect_input = user_choice.next();
                if (collect_input.matches("[0-9]+")) {
                    if (Integer.parseInt(collect_input) != 0) {
                        set.add(Integer.parseInt(collect_input));
                        System.out.print("Enter Next Number:  ");
                    } else {
                        JOptionPane.showMessageDialog(null,
                                "You cannot enter zero. Repeat that!\nType 'done' when you're finished.");
                    }
                } else if (!"DONE".equals(collect_input.toUpperCase())) {
                    JOptionPane.showMessageDialog(null,
                            "Wrong Input. Repeat that!\nType 'done' when you're finished.");
                }
            } while (!"DONE".equals(collect_input.toUpperCase()));
        } catch (NumberFormatException | HeadlessException e) {
            System.out.println(e.getMessage());
        }
        
        String render_result = "The HCF of ";
        for (int number : set) {
            render_result += number + "; ";
        }
        render_result += "is:  ";

        h_c_f = new FastHCF(set);
        System.out.println(render_result + h_c_f.getHCF());
    }

}

File: Arithmetic_HCFInput.cs
using System;
using System.Collections.Generic;
using System.Text.RegularExpressions;

namespace Arithmetic_CS
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Welcome to our demonstration sequels");
            Console.WriteLine("Hope you enjoy (and follow) the lessons.");
            Console.WriteLine("\r\n");

            /*
            * Collect input.
             */
            // Collect input
            string collect_input; // For collecting user input

            List<int> set = new List<int>();

            Console.WriteLine("Welcome to our Find H.C.F. program.");
            Console.WriteLine("Enter your series of number for H.C.F.");
            Console.WriteLine("Type 'done' when you are through with entering your numbers.\n");

            string collect_input_text = "Enter First Number:  ";
            try
            {
                do
                {
                    // Get keyboard input
                Console.Write(collect_input_text);
                    collect_input = Console.ReadLine();
                    if (Regex.IsMatch(collect_input, @"[0-9]+"))
                    {
                        if (int.Parse(collect_input) != 0)
                        {
                            set.Add(int.Parse(collect_input));
                            collect_input_text = "Enter Next Number:  ";
                        }
                        else
                        {
                            Console.WriteLine("You cannot enter zero. Repeat that!Type 'done' when you're finished.");
                        }
                    }
                    else if (!"DONE".Equals(collect_input.ToUpper()))
                    {
                        Console.WriteLine("Wrong Input. Repeat that!\r\n Type 'done' when you're finished.");
                    }
                } while (!"DONE".Equals(collect_input.ToUpper()));
            }
            catch (Exception e) {
                Console.WriteLine(e.Message);
            }

            FastHCF h_c_f = new FastHCF(set);
            Console.WriteLine("The H.C.F. of " + String.Join(", ", set) + " is " + h_c_f.getHCF());

        }
    }
}
